Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebContext.cpp

```diff
@@ -128,9 +128,15 @@ class URISchemeTest: public LoadTrackingTest {
 
         const char* method = webkit_uri_scheme_request_get_http_method(request);
         g_assert_nonnull(method);
-        if (!g_strcmp0(scheme, "post"))
+        if (!g_strcmp0(scheme, "post")) {
             g_assert_cmpstr(method, ==, "POST");
-        else
+
+            GInputStream* body = webkit_uri_scheme_request_get_http_body(request);
+            g_assert_nonnull(body);
+            char readBuffer[8] = { 0 };
+            g_input_stream_read_async(body, readBuffer, 8, G_PRIORITY_DEFAULT, NULL, NULL, NULL);
+            g_assert_cmpstr(readBuffer, ==, "X-Test=A");
```

Comment:
This works because the stream is pollable and it can read immediately, but you should use `g_input_stream_read_all` instead.